packages/vm/src/index.ts

```diff
@@ -117,6 +117,10 @@ export interface VMOpts {
    * pointing to a Shanghai block: this will lead to set the HF as Shanghai and not the Merge).
    */
   hardforkByTD?: BNLike
+  /**
+   * Limit parameter returns to the given hardforks
+   */
+  supportedHardforks?: Array<string | Hardfork>
```

Comment:
Ah no, we do not want to "recreate" this as a feature here, so you can remove this again.

The only reason we have this in the VM is that `Common` sometimes temporarily supports more HFs than implemented in the VM (e.g. if we already add the Merge HF).

So you can keep the `const supportedHardforks = [ ... ]` array from below and then the only thing is you need to do:

1. Wait for `Common` to be initialized inside the VM (also: below)
2. Check if the HF from Common (which might be passed in) is present in the `supportedHardforks` array and otherwise throw

Some background on this: this whole `supportedHardforks` thing had more reason to be there in the times where we didn't support the full range of HFs yet with the VM (so didn't yet backport the old HFs). Now it would even be an option to remove this check completely, but I guess for these temporary edge cases with new not-applied HFs it should still make some sense.

Reply:
Ah ok, that makes sense. PR updated per your feedback. I updated the test case and the VM README as well.

One question: when I used the `supportedHardforks` array that was already in the code, about 20 VM API tests failed because `London` was not included in the `supportedHardforks` array. I added `London` to the array and all the tests passed. Which made me think, I noticed there was another item in the breaking release issue to update the default hardfork to `London`. Should I include that change in this PR? Or I could tackle it separately in a different PR (or someone else could).